traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -273,3 +301,369 @@ RETURNING id, last_updated`
+		return
+	}
+	lastUpdated := tc.TimeNoMod{}
+	for rows.Next() {
+		if err := rows.Scan(&lastUpdated); err != nil {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("scanning lastUpdated from role update: %w", err))
+			return
+		}
+		roleV4.LastUpdated = &lastUpdated.Time
+	}
+
+	if roleCapabilities != nil && *roleCapabilities != nil {
+		userErr, sysErr, errCode = deleteRoleCapabilityAssociations(inf.Tx, roleID)
+		if userErr != nil || sysErr != nil {
+			api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+			return
+		}
+		userErr, sysErr, errCode = createRoleCapabilityAssociations(inf.Tx, roleID, roleCapabilities)
+		if userErr != nil || sysErr != nil {
+			api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+			return
+		}
+	}
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "role was updated.")
+	var roleResponse interface{}
+	var capabilities []string
+	if roleCapabilities != nil {
+		capabilities = *roleCapabilities
+	}
+	roleResponse = tc.RoleV4{
+		Name:        roleName,
+		Permissions: capabilities,
+		Description: roleDesc,
+	}
+	api.WriteAlertsObj(w, r, http.StatusOK, alerts, roleResponse)
+	changeLogMsg := fmt.Sprintf("ROLE: %s, ID: %d, ACTION: Updated Role", roleName, roleID)
+	api.CreateChangeLogRawTx(api.ApiChange, changeLogMsg, inf.User, tx)
+}
+
+func deleteRoleQuery() string {
+	return `DELETE FROM role WHERE id = $1`
+}
+
+func readQuery() string {
+	return `SELECT
+id,
+name,
+description,
+priv_level,
+last_updated,
+ARRAY(SELECT rc.cap_name FROM role_capability AS rc WHERE rc.role_id=id) AS permissions
+FROM role`
+}
+
+func createQuery() string {
+	return `INSERT INTO role (
+name,
+description,
+priv_level
+) VALUES (
+$1,
+$2,
+$3
+)
+RETURNING id, last_updated`
+}
+
+func createRoleCapabilityAssociations(tx *sqlx.Tx, roleID int, permissions *[]string) (error, error, int) {
+	result, err := tx.Exec(associateCapabilities(), roleID, pq.Array(permissions))
+	if err != nil {
+		return nil, fmt.Errorf("creating role capabilities: %w", err), http.StatusInternalServerError
+	}
+
+	if rows, err := result.RowsAffected(); err != nil {
+		return nil, fmt.Errorf("could not check result after inserting role_capability relations: %w", err), http.StatusInternalServerError
+	} else if expected := len(*permissions); int(rows) != expected {
+		return fmt.Errorf("wrong number of role_capability rows created: %d expected: %d", rows, expected), nil, http.StatusBadRequest
+	}
+	return nil, nil, http.StatusOK
+}
+
+func deleteRoleCapabilityAssociations(tx *sqlx.Tx, roleID int) (error, error, int) {
+	result, err := tx.Exec(deleteAssociatedCapabilities(), roleID)
+	if err != nil {
+		return nil, fmt.Errorf("deleting role capabilities: %w", err), http.StatusInternalServerError
+	}
+	if _, err := result.RowsAffected(); err != nil {
+		return nil, fmt.Errorf("could not check result after inserting role_capability relations: %w", err), http.StatusInternalServerError
+	}
```

Comment:
but actually it doesn't really seem to be checking the resulting rows affected, since that value is thrown away to `_` - should it be doing something with the real information returned by this call?

Reply:
Actually, it shouldn't matter if the affected rows is 0 or more than 0, hence, I think we don't need this check anymore at all.